bindings/bindingtester/tests/directory_state_tree.py

```diff
@@ -0,0 +1,266 @@
+import sys
+
+class TreeNodeState:
+    def __init__(self, dir_id, is_directory, is_subspace, has_known_prefix, root, is_partition):
+        self.dir_id = dir_id
+        self.root = root
+        self.is_directory = is_directory
+        self.is_subspace = is_subspace
+        self.has_known_prefix = has_known_prefix
+        self.children = {}
+        self.deleted = False
+        self.is_partition = is_partition
+
+class Pointer:
+    def __init__(self, obj):
+        self.obj = obj
+
+    def get(self):
+        return self.obj
+
+    def set(self, obj):
+        self.obj = obj
+
+# Represents an element of the directory hierarchy, which could have multiple states 
```

Comment:
I was trying to add the ending clause as an extra piece of info rather than have it be part of what's represented (i.e. this represents "an element of the directory hierarchy", not "an element of the directory hierarchy which could have multiple states". I can reword this though.

Reply:
Hm, I see. I suppose it's fine, then.